dirty_equals/_other.py

```diff
@@ -369,3 +373,166 @@ def equals(self, other: Any) -> bool:
                 return False
 
         return True
+
+
+class IsDataclassType(DirtyEquals[Any]):
+    """
+    Checks that an object is a dataclass type.
+
+    Inherits from [`DirtyEquals`][dirty_equals.DirtyEquals].
+
+    ```py title="IsDataclassType"
+    from dataclasses import dataclass
+    from dirty_equals import IsDataclassType
+
+    @dataclass
+    class Foo:
+        a: int
+        b: int
+
+    foo = Foo(1, 2)
+
+    assert Foo == IsDataclassType
+    assert foo != IsDataclassType
+    ```
+    """
+
+    def equals(self, other: Any) -> bool:
+        return is_dataclass(other) and isinstance(other, type)
+
+
+class IsDataclass(DirtyEquals[Any]):
+    """
+    Checks that an object is an instance of a dataclass.
+
+    Inherits from [`DirtyEquals`][dirty_equals.DirtyEquals] and it can be initialised with specific keyword arguments to
+    check exactness of dataclass fields, by comparing the instance `__dict__`  with [`IsDict`][dirty_equals.IsDict].
+    Moreover it is possible to check for strictness and partialness of the dataclass, by setting the `strict` and
+    `partial` attributes using the `.settings(strict=..., partial=...)` method.
+
+    Remark that passing no kwargs to `IsDataclass` initialization means fields are not checked, not that the dataclass
+    is empty, namely `IsDataclass()` is the same as `IsDataclass`.
+
+    ```py title="IsDataclass"
+    from dataclasses import dataclass
+    from dirty_equals import IsInt, IsDataclass
+
+    @dataclass
+    class Foo:
+        a: int
+        b: int
+        c: str
+
+    foo = Foo(1, 2, 'c')
+
+    assert foo == IsDataclass
+    assert foo == IsDataclass(a=IsInt, b=2, c='c')
+    assert foo == IsDataclass(b=2, a=1).settings(partial=True)
+    assert foo != IsDataclass(a=IsInt, b=2).settings(strict=True)
+    assert foo == IsDataclass(a=IsInt, b=2).settings(strict=True, partial=True)
+    assert foo != IsDataclass(b=2, a=1).settings(strict=True, partial=True)
+    ```
+    """
+
+    def __init__(self, **fields: Any):
+        """
+        Args:
+            fields: key-value pairs of field-value to check for.
+        """
+        self.strict = False
+        self.partial = False
+        self._post_init()
+        super().__init__(**fields)
+
+    def _post_init(self) -> None:
+        pass
+
+    def equals(self, other: Any) -> bool:
+        if is_dataclass(other) and not isinstance(other, type):
+            if self._repr_kwargs:
+                return self._fields_check(other)
+            else:
+                return True
+        else:
+            return False
+
+    def settings(
```

Comment:
Notice that with this implementation it is possible to do something like:

```py
from dirty_equals import IsPartialDict

IsPartialDict(...).settings(partial=False)
```
Not sure if this is intended behaviour. Maybe I can work on a different Issue/PR to check for it.

Reply:
doesn't sound too bad to me.